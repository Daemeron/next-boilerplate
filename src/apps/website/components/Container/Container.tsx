import React, { ReactChild } from 'react';
import styled from 'styled-components';

import { ThemePropsInterface } from '../../core/theme/Theme';

interface ContainerProps {
  children?: ReactChild | ReactChild[];
  theme?: ThemePropsInterface;
}

export const Container = (props: ContainerProps): JSX.Element => (
  <ContainerBox theme={props.theme}>{props.children}</ContainerBox>
);

export const ContainerBox = styled.div<ThemePropsInterface>`
  max-width: 100%;
  margin: 0 auto;
  padding: 0 24px;
  @media (min-width: 576px) {
    max-width: 576px;
  }
  @media (min-width: 768px) {
    max-width: 768px;
    padding: 0 32px;
  }
  @media (min-width: 992px) {
    max-width: 992px;
    padding: 0 48px;
  }
  @media (min-width: 1200px) {
    max-width: 1200px;
    padding: 0 48px;
  }
`;
