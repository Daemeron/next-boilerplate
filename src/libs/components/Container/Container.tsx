import React, { ReactChild } from 'react';
import styled from 'styled-components';

interface ContainerProps {
  children?: ReactChild | ReactChild[];
}

export const Container = (props: ContainerProps): JSX.Element => (
  <ContainerBox>{props.children}</ContainerBox>
);

export const ContainerBox = styled.div``;
