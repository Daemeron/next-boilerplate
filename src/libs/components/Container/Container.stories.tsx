import React, { Fragment } from 'react';
import styled from 'styled-components';

import { Container } from './Container';

export default {
  title: 'Components/Container',
};

export const WithText = (): JSX.Element => (
  <Fragment>
    <Container>
      <Paragraph>dummy text</Paragraph>
    </Container>
  </Fragment>
);

const Paragraph = styled.p``;
