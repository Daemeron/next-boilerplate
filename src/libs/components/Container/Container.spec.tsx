import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import { Container } from './Container';

describe('Container [style=default]', () => {
  test('matches the snapshot', () => {
    const container = render(<Container />);
    expect(container.container).toMatchSnapshot();
  });
});
