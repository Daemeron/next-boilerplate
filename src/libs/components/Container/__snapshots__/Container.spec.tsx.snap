// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container [style=default] matches the snapshot 1`] = `
<div>
  <div
    class=""
  />
</div>
`;
